src/core/i18n/i18n.ts

```diff
@@ -0,0 +1,19 @@
+import { getRequestConfig } from "next-intl/server";
+
+import { defaultLocale, locales } from "./config";
+import { getLocale } from "./locale";
+import type { Locale } from "./types";
+
+const i18nRequestConfig = getRequestConfig(async () => {
+  const locale = await getLocale() as Locale;
+
+  return {
+    locale,
+    messages:
+      locale === defaultLocale || !locales.includes(locale)
+        ? (await import("@public/locales/en.json")).default
```

Comment:
I would add the default locale here instead of using english as fallback as such

```
(await import(`@public/locales/${defaultLocale}.json`)).default
```

Reply:
Changed it, thank you for the clarification.